omnipaxos/src/sequence_paxos/follower.rs

```diff
@@ -16,67 +16,83 @@ where
+                            .expect(READ_ERROR_MSG);
+                        (delta_snapshot, suffix, compacted_idx)
+                    } else {
+                        let suffix = self
+                            .internal_storage
+                            .get_suffix(prep.decided_idx)
+                            .expect(READ_ERROR_MSG);
+                        (None, suffix, prep.decided_idx)
+                    };
+                Some(LogSync {
+                    decided_snapshot,
+                    suffix,
+                    sync_idx,
+                    stopsign,
+                })
             } else if na == prep.n_accepted && accepted_idx > prep.accepted_idx {
-                let compacted_idx = self.internal_storage.get_compacted_idx();
-                if T::Snapshot::use_snapshots() && compacted_idx > prep.accepted_idx {
-                    let (delta_snapshot, _) = self
-                        .internal_storage
-                        .create_diff_snapshot(prep.decided_idx)
-                        .expect("storage error while trying to read diff snapshot");
-                    let suffix = self
-                        .internal_storage
-                        .get_suffix(decided_idx)
-                        .expect("storage error while trying to read log suffix");
-                    (delta_snapshot, suffix)
-                } else {
-                    let suffix = self
-                        .internal_storage
-                        .get_suffix(prep.accepted_idx)
-                        .expect("storage error while trying to read log suffix");
-                    (None, suffix)
-                }
+                // I'm more up to date and no split-brain possible => append onto preparer's
+                // accepted entries.
+                let (decided_snapshot, suffix, sync_idx) =
+                    if T::Snapshot::use_snapshots() && decided_idx > prep.accepted_idx {
+                        // Note: We snapshot from preparer's decided and not preparer's accepted because
+                        // snapshots currently can't handle merging onto accepted entries.
+                        let (delta_snapshot, compacted_idx) = self
+                            .internal_storage
+                            .create_diff_snapshot(prep.decided_idx)
+                            .expect(READ_ERROR_MSG);
+                        let suffix = self
+                            .internal_storage
+                            .get_suffix(decided_idx)
+                            .expect(READ_ERROR_MSG);
+                        (delta_snapshot, suffix, compacted_idx)
+                    } else {
+                        let suffix = self
+                            .internal_storage
+                            .get_suffix(prep.accepted_idx)
+                            .expect(READ_ERROR_MSG);
+                        (None, suffix, prep.accepted_idx)
```

Comment:
This logic is more or less the same as in the previous case but with a different index used to get the suffix. Can we create a function for it?